setup.py

```diff
@@ -3,65 +3,72 @@
 
 from setuptools import setup, Extension
 
-try:
-  from Cython.Distutils import build_ext
-except ImportError:
-  print("""Cannot find Cython!
-  Cython is required to correctly build pyBloomFilter's C extensions.
-  In most cases, running the following command should be sufficient:
-    $ pip install Cython
-
-  Exception: ImportError
-  """)
-  exit()
+if sys.version_info[0] < 3:
+    raise SystemError("This package is for Python version 3 and above.")
 
 here = os.path.dirname(__file__)
+
 # Get the long description from the README file
-with open(os.path.join(here, 'README.markdown'), encoding='utf-8') as fp:
-  long_description = fp.read()
+with open(os.path.join(here, "README.markdown"), encoding="utf-8") as fp:
+    long_description = fp.read()
 
+setup_kwargs = {}
 
 ext_files = [
-  'src/mmapbitarray.c',
-  'src/bloomfilter.c',
-  'src/md5.c',
-  'src/primetester.c',
-  'src/MurmurHash3.c',
-  'src/pybloomfilter.pyx'
+    "src/mmapbitarray.c",
+    "src/bloomfilter.c",
+    "src/md5.c",
+    "src/primetester.c",
+    "src/MurmurHash3.c"
 ]
 
-print("info: Building from Cython")
+try:
+    if "--no-cython" in sys.argv:
+        sys.argv.remove("--no-cython")
+        raise ImportError()
+
+    from Cython.Distutils import build_ext
+
+    ext_files.append("src/pybloomfilter.pyx")
+    setup_kwargs["cmdclass"] = {"build_ext": build_ext}
+    print("Building using Cython")
+except ImportError:
```

Comment:
This can be very ambiguous, since `ImportError` can be either the one you raise in line 28, or from 30.

Don't use exceptions for case-matching. Instead branch out. A refactor block is below:

```py
# branch out based on `--no-cython` in `argv`.
# assume `--cython` as default to avoid havig to deal with both params being there.

if "--no-cython" in sys.argv:
	# note that we let the exception bubble up if `pybloomfilter.c` doesnt exist.
	ext_files.append("src/pybloomfilter.c")
else:
	from Cython.Distutils import build_ext

	ext_files.append("src/pybloomfilter.pyx")
	setup_kwargs["cmdclass"] = {"build_ext": build_ext}
```

Reply:
You're right, it makes sense - I have mostly left this code as is when I pushed the changes.

Reply:
ca3d656